libraries/chain/account_evaluator.cpp

```diff
@@ -117,9 +117,15 @@ void verify_account_votes( const database& db, const account_options& options )
          }
       }
    }
+   if(account != account_id_type(0) &&
+         (options.votes != account(db).options.votes || options.voting_account != account(db).options.voting_account)) {
+      auto &stats_obj = db.get_account_stats_by_owner(account);
+      db.modify(stats_obj, [&](account_statistics_object &obj) {
+         obj.last_vote_time = db.head_block_time();
+      });
+   }
```

Comment:
This means every time when `do_evaluate` is called (E.G. by `verify_transaction` API ?), the data will be updated. But `do_evaluate` can fail.

Reply:
Usually as an unwritten principle we don't modify database in `do_evaluate`, but probably it's practically fine to do so. @pmconrad your opinion?

Reply:
I agree that `do_evaluate` must not modify the database.
IMO it's a good design decision to separate the various stages of operation evaluation, and we should stick with it.

Reply:
got it, thanks. ill refactor to do the update in `do_apply` or something different.